Blazor partial class support (#15221)
aspnet/AspNetCore.Docs: aspnetcore/blazor/components.md

* C# code is placed in a code-behind file defined as a partial class.

The following example shows the default `Counter` component with an `@code` block in an app generated from a Blazor template. HTML markup, Razor code, and C# code are in the same file:

*Counter.razor*:

```cshtml
@page "/counter"

<h1>Counter</h1>

<p>Current count: @currentCount</p>

<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>

@code {
    int currentCount = 0;

    void IncrementCount()
    {
        currentCount++;
    }
}
```

The `Counter` component can also be created using a code-behind file with a partial class:

*Counter.razor*:

```cshtml
@page "/counter"

<h1>Counter</h1>

<p>Current count: @currentCount</p>

<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>
```

*Counter.razor.cs*:

```csharp
namespace BlazorApp.Pages
{
    public partial class Counter
    {
        int currentCount = 0;

        void IncrementCount()
        {
            currentCount++;
        }
    }
}
```

::: moniker-end

::: moniker range="< aspnetcore-3.1"

> ## Base class inheritance for a "code-behind" experience

Reviewer: I think we should rename and reword this section to remove the emphasis on using inheritance for code-behind. If you want a code-behind file, you should use partial classes, but we still support defining a base class for components.

```suggestion
## Specifying a component base class
```

> Component files mix HTML markup and C# processing code in the same file. The `@inherits` directive can be used to provide Blazor apps with a "code-behind" experience that separates component markup from processing code.

Reviewer: ```suggestion
The `@inherits` directive can be used to specify a base class for a component.
```
